scripts/generating_yaml.py

```diff
@@ -0,0 +1,213 @@
+
+def j(USBhub_SN):
+    return USBhub_SN
+
+def k(camera_into_USBhub_port):
+    return camera_into_USBhub_port
+
+
+# In[8]:
+
+birger_sn = data['birger_SN']
+birger_serial_number = interactive(a, birger_serial_number=birger_sn);
+sbig_sn = data['sbig_SN']
+sbig_serial_number = interactive(b, sbig_serial_number=sbig_sn);
+lens_sn = data['lens_SN']
+lens_serial_number = interactive(c, lens_serial_number=lens_sn);
+filter_ID = data['filter_ID']
+filter_ID_code = interactive(d, filter_ID_code=filter_ID);
+serial_into_USBhub = data['serial_into_USBhub_port']
+serial_into_USBhub_port = interactive(e, serial_into_USBhub_port=serial_into_USBhub);
+camera_into_serial = data['camera_into_serial_port']
+camera_into_serial_port = interactive(f, camera_into_serial_port=camera_into_serial);
+USBhub = data['USBhub_SN']
+USBhub_SN = interactive(j, USBhub_SN=USBhub);
+camera_into_USBhub = data['camera_into_USBhub_port']
+camera_into_USBhub_port = interactive(k, camera_into_USBhub_port=camera_into_USBhub);
+
+
+# In[9]:
+
+def h(add_device):
+    birger_SN = birger_serial_number.result
+    sbig_SN = sbig_serial_number.result
+    lens_SN = lens_serial_number.result
+    filter_ID = filter_ID_code.result
+    serial_to_USBhub_port = serial_into_USBhub_port.result
+    camera_to_serial_port = camera_into_serial_port.result
+    USB_hub_SN = USBhub_SN.result
+    camera_to_USBhub_port = camera_into_USBhub_port.result
+    additional_device = {'model' : 'sbig',
+                                                    'port' : sbig_SN,
+                                                   'filter_type' : filter_ID,
+                                                    'focuser' : {'model' : 'birger',
+                                                                  'port' : birger_SN
+
+                                                    },
+                                                    'lens' : {'model' : 'canon',
+                                                               'port' : lens_SN},
+                                                    'USB_hub_serial_number' : USB_hub_SN,
+                                                    'camera_into_serial_adaptor_port' : camera_to_serial_port,
+                                                    'serial_adaptor_into_USBhub_port' : serial_to_USBhub_port,
+                                                   'camera_into_USBhub_port' : camera_to_USBhub_port
+                                                              }
+    if data_list['cameras']['devices'] == [None]:
+        data_list['cameras']['devices'] = [additional_device]
+    else:
+        data_list['cameras']['devices'].append(additional_device)
+    return (add_device)
+
+
+# In[10]:
+
+date_info = datetime.datetime.today()
+datetime_str = date_info.strftime('%Y_%m_%d_%H_%M')
+
+archive_filename = '{}_{}.{}'.format('huntsman', datetime_str, 'yaml')
+
+
+# In[11]:
+
+archive_filename
+
+
+# In[12]:
+
+def g(create_file):
```

Comment:
What does create_file do? You don't seem to use it again though it's returned. Doesn't seem to be used in the function either? Or do I not understand whats happening :P

Reply:
I think it's a bit hard to see what's happening because you aren't seeing the jupyter widgets that come up when it is run. The widgets work in an odd way, the function is defining what to do once the widget is clicked. So the function doesn't have a specific arg value, it just returns what the widget outputs. I'll try and explain it a bit more in the code if I can, too.

I think the return line came about from the earlier model I was following for those functions, but I have just run it after removing the return and it appears to be working fine, which does make sense.

Reply:
Okidoke, yeah I'm not very familiar with how the widgets work